mm2src/coins/tendermint/tendermint_coin.rs

```diff
@@ -2195,6 +2214,128 @@ impl MmCoin for TendermintCoin {
+
+        loop {
+            let node_uri = match self.rpc_client().await {
+                Ok(client) => client.uri(),
+                Err(e) => {
+                    error!("{e}");
+                    continue;
+                },
+            };
+
+            let socket_address = format!("{}/{}", http_uri_to_ws_address(node_uri), "websocket");
+
+            let mut wsocket = match tokio_tungstenite_wasm::connect(socket_address).await {
+                Ok(ws) => ws,
+                Err(e) => {
+                    error!("{e}");
+                    continue;
+                },
+            };
+
+            // Filter received TX events
+            if let Err(e) = wsocket.send(receiver_q.clone()).await {
+                error!("{e}");
+                continue;
+            }
+
+            // Filter spent TX events
+            if let Err(e) = wsocket.send(spender_q.clone()).await {
+                error!("{e}");
+                continue;
+            }
+
+            while let Some(message) = wsocket.next().await {
+                let msg = match message {
+                    Ok(tokio_tungstenite_wasm::Message::Text(data)) => data.clone(),
+                    Ok(tokio_tungstenite_wasm::Message::Close(_)) => break,
+                    Err(err) => {
+                        error!("{err}");
+                        break;
+                    },
+                    _ => continue,
+                };
+
+                if let Ok(json_val) = json::from_str::<json::Value>(&msg) {
+                    let transfers: Vec<String> =
+                        json::from_value(json_val["result"]["events"]["transfer.amount"].clone()).unwrap_or_default();
+
+                    let mut denoms: Vec<String> = transfers
+                        .iter()
+                        .map(|t| {
+                            let amount: String = t.chars().take_while(|c| c.is_numeric()).collect();
+                            let denom = &t[amount.len()..];
+                            denom.to_owned()
+                        })
+                        .collect();
+
+                    denoms.dedup();
+                    drop_mutability!(denoms);
+
+                    for denom in denoms {
+                        if let Some((ticker, decimals)) = self.active_ticker_and_decimals_from_denom(&denom) {
+                            let balance_denom = match self.account_balance_for_denom(&self.account_id, denom).await {
+                                Ok(balance_denom) => balance_denom,
+                                Err(e) => {
+                                    error!("{e}");
+                                    continue;
+                                },
+                            };
+
+                            let balance_decimal = big_decimal_from_sat_unsigned(balance_denom, decimals);
+
+                            // Only broadcast when balance is changed
+                            let mut broadcast = false;
```

Comment:
Do we really need to broadcast only if balance was changed? Can we assume that if the transfer happened, then the balance is changed for sure?

Reply:
No and yes. We only want to broadcast when data is changed. When transfer happened, the balance is changed for sure. I was thinking to add calling my_balance every 20-30 second here(to make sure we have the correct balance when we miss events from socket), in that case this was going to be good check.